### PR TITLE
(tests): Enables integration tests including Pulsar 

### PR DESCRIPTION
**Pull request by therapon**
Closes: #205 

* CI no longer builds integration test binary 
    * this is actually causing failures due to how we re-upload older artifacts when skipping. This PR should fix the failures for now but we need a better way to donwload older artifacts on skip
* CI steps for integration tests are now using `Makefile` 
* New Docker file for integration steps 
* New docker compose file for starting pulsar and kaskada 

Sample run that I manually initiated on the branch
https://github.com/kaskada-ai/kaskada/actions/runs/4621121066

### CODE REVIEW

**Review by rltvty — APPROVED** [review]

File: tests/integration/docker-compose-ci-integration.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      retries: 5
+      start_period: 20s
+    # restart: unless-stopped
+
+  kaskada:
+    build:
+      context: ../../
+      dockerfile: Dockerfile.integration
+    # image: tsk-local-integ:latest
+    # container_name: kaskada
```
Comment (jordanrfrazier) [inline tests/integration/docker-compose-ci-integration.yml:42]: .

File: tests/integration/docker-compose-ci-integration.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    networks:
+      - integration
+    command: [ "bin/pulsar", "standalone" ]
+    healthcheck:
+      test: curl --fail  http://localhost:8080/admin/v2/brokers/health
+      interval: 60s
+      timeout: 10s
+      retries: 5
+      start_period: 20s
+    # restart: unless-stopped
```
Comment (jordanrfrazier) [inline tests/integration/docker-compose-ci-integration.yml:35]: .

File: .github/workflows/ci_engine.yml [hunk 3]
```diff
@@ -29,6 +29,24 @@ env:
   IMAGE: ghcr.io/${{ github.repository }}/engine-integration-tests
 
 jobs:
+
+  integration_skip:
+    name: Integration Tests Skip Duplicates
+    continue-on-error: true # Uncomment once integration is finished
```
Comment (jordanrfrazier) [inline .github/workflows/ci_engine.yml:35]: Do we need to remove this?

**Review by jordanrfrazier — APPROVED** [review]

File: .github/workflows/ci_engine.yml (hunk 3 above)
Comment (therapon) [reply .github/workflows/ci_engine.yml:35]: That I am not sure ... let me check.

File: tests/integration/docker-compose-ci-integration.yml (hunk 2 above)
Comment (therapon) [reply tests/integration/docker-compose-ci-integration.yml:35]: Will remove.

File: tests/integration/docker-compose-ci-integration.yml (hunk 1 above)
Comment (therapon) [reply tests/integration/docker-compose-ci-integration.yml:42]: rats ... I always miss something 
Will remove.

**Pull request merged**